packages/backend/src/util/request.ts

```diff
@@ -37,10 +37,23 @@ export function checkCrossOrigin({
   return false;
 }
 
-export function convertHostHeaderValueToURL(host: string): URL {
+export function convertHostHeaderValueToURL(host: string, protocol = 'https:'): URL {
   /**
    * The protocol is added for the URL constructor to work properly.
    * We do not check for the protocol at any point later on.
    */
-  return new URL(`https://${host}`);
+  return new URL(`${protocol}//${host}`);
+}
+
+const PROTOCOL_TO_PORT_MAPPING: Record<string, string> = {
```

Comment:
🙃 No need to explicitly type as this is using `as const`
```suggestion
const PROTOCOL_TO_PORT_MAPPING = {
```

Reply:
Removing the type causes a TS error.
<img width="909" alt="Screenshot 2023-05-31 at 11 49 35" src="https://github.com/clerkinc/javascript/assets/3936408/9317eb7f-4912-4651-a7a3-47e41ed0b752">
<img width="745" alt="Screenshot 2023-05-31 at 11 49 29" src="https://github.com/clerkinc/javascript/assets/3936408/5ea1c7f5-5b6c-4ab7-a301-db3faff72785">